src/api/options-lifecycle.md

```diff
@@ -1,174 +1,174 @@
 
-- **Type**
+- **Loại**
 
   ```ts
   interface ComponentOptions {
     beforeMount?(this: ComponentPublicInstance): void
   }
   ```
 
-- **Details**
+- **Chi tiết**
 
-  When this hook is called, the component has finished setting up its reactive state, but no DOM nodes have been created yet. It is about to execute its DOM render effect for the first time.
+  Khi hook này được gọi, component đã hoàn thành việc thiết lập reactive state, nhưng chưa có DOM nodes nào được tạo ra. Nó sẽ thực thi DOM render effect lần đầu tiên.
 
-  **This hook is not called during server-side rendering.**
+  **Hook này không được gọi trong server-side rendering.**
 
 ## mounted {#mounted}
 
-Called after the component has been mounted.
+Được gọi sau khi component đã được mounted.
 
-- **Type**
+- **Loại**
 
   ```ts
   interface ComponentOptions {
     mounted?(this: ComponentPublicInstance): void
   }
   ```
 
-- **Details**
+- **Chi tiết**
 
-  A component is considered mounted after:
+  Một component được coi là mounted sau khi:
 
-  - All of its synchronous child components have been mounted (does not include async components or components inside `<Suspense>` trees).
+  - Tất cả các component con đồng bộ của nó đã được mounted (không bao gồm các component bất đồng bộ hoặc component trong cây `<Suspense>`).
 
-  - Its own DOM tree has been created and inserted into the parent container. Note it only guarantees that the component's DOM tree is in-document if the application's root container is also in-document.
+  - Cây DOM của nó đã được tạo và được chèn vào container cha. Lưu ý rằng nó chỉ đảm bảo rằng cây DOM của component nằm trong document nếu root container của application cũng nằm trong document.
 
-  This hook is typically used for performing side effects that need access to the component's rendered DOM, or for limiting DOM-related code to the client in a [server-rendered application](/guide/scaling-up/ssr).
+  Hook này thường được sử dụng để thực hiện các side effects, những cái mà cần truy cập vào DOM của component đã được render, hoặc để giới hạn code liên quan đến DOM chỉ chạy trên client trong một [server-rendered application](/guide/scaling-up/ssr).
 
-  **This hook is not called during server-side rendering.**
+  **Hook này không được gọi trong server-side rendering.**
 
 ## beforeUpdate {#beforeupdate}
 
-Called right before the component is about to update its DOM tree due to a reactive state change.
+Được gọi ngay trước khi component chuẩn bị cập nhật DOM tree của nó do một reactive state change.
 
-- **Type**
+- **Loại**
 
   ```ts
   interface ComponentOptions {
     beforeUpdate?(this: ComponentPublicInstance): void
   }
   ```
 
-- **Details**
+- **Chi tiết**
 
-  This hook can be used to access the DOM state before Vue updates the DOM. It is also safe to modify component state inside this hook.
+  Hook này có thể được sử dụng để truy cập vào DOM state trước khi Vue cập nhật DOM. Cũng an toàn khi thay đổi component state bên trong hook này.
```

Comment:
DOM, và cũng an toàn khi thay đổi component state bên trong hook này.
DOM. Việc thay đổi component state trong hook này cũng an toàn.

-> Câu ghép hoặc tách thành câu riêng.